### ISSUE TITLE
Unlock configuration file during download

### ISSUE THREAD

**Issue by LilithHafner** [JuliaLang/juliaup#435]
When I run `juliaup add 1.7` it takes a long time to download because of a slow internet connection. This is okay, but when I try to launch `julia +1.8.2`  in a different terminal, I get "Juliaup configuration is locked by another process, waiting for it to unlock." It would be nice to unlock the configuration during the download process and only lock it very briefly when doing actual work with the configuration.

**Comment by StefanKarpinski** [comment]
Might also be possible to do more granular read/write locking: many read locks can co-exist and they only conflict with a write lock, which is exclusive. I have no idea if Rust supports that, but if it does that would be ideal. If the operations expect the configuration not to change while they're doing their thing, that might be a better approach than making the lock region smaller.

**Comment by LilithHafner** [comment]
Agreed, however, in step three, instead of checking for any change I'd check to see if a new installation is still warranted (same check that is performed in step 1 to decide whether to tell the user "Already up to date"). That way it is possible to install multiple different versions in parallel and requires less memory (the former is substantive, the latter is not)

**Comment by davidanthoff** [comment]
Yes, that would be even better!

**Comment by evetion** [comment]
Relates to #561

**Comment by davidanthoff** [comment]
I'm working on this one right now.

**Comment by LilithHafner** [comment]
Thanks! Feel free to @ me when you make a PR, I'd love to see it.

**Comment by davidanthoff** [comment]
@gaurav-arya the issue you are describing is not related to the topic of this issue here, the problem you have is tracked in https://github.com/JuliaLang/juliaup/issues/561.

**Comment by stevengj** [comment]
Any hope of progress on this?

**Comment by IanButterworth** [comment]
I asked claude to factor-in all the discussions and open PRs: https://github.com/JuliaLang/juliaup/pull/1517

**Comment by bjarthur** [comment]
is there a hack to fix it?  where is the lock file that i could manually delete?

**Comment by bjarthur** [comment]
`% rm ~/.julia/juliaup/.juliaup-lock` !!!
